current_fall2025/scripts/RAG/query_enhancement.py

```
@@ -0,0 +1,82 @@
+    # Remove markdown code fences if present
+    if content.startswith("```"):
+        content = re.sub(r"^```[a-zA-Z0-9]*\n?", "", content)
+        content = re.sub(r"```$", "", content)
+        content = content.strip()
+
+    try:
+        data = json.loads(content)
+        parsed = QueryRewrite(**data)
+        final_q = f"{parsed.improved_query.strip()} {parsed.expanded_query.strip()}".strip()
+        logging.info(f"✅ Query rephrased in {time.time() - start:.2f}s: '{final_q}'")
+        return final_q
```

Ruff (0.14.5)

Consider moving this statement to an `else` block

(TRY300)